mypy_json_report.py

```diff
@@ -54,21 +71,67 @@ def main() -> None:
         type=pathlib.Path,
         help="The file to write the JSON report to. If omitted, the report will be written to STDOUT.",
     )
+    parse_parser.add_argument(
+        "-d",
+        "--diff-old-report",
+        default=None,
+        type=pathlib.Path,
+        help=textwrap.dedent(
+            f"""\
+            An old report to compare against. We will compare the errors in there to the new report.
+            Fail with return code {ErrorCodes.ERROR_DIFF} if we discover any new errors.
+            New errors will be printed to stderr.
+            Similar errors from the same file will also be printed
+            (because we don't know which error is the new one).
+            For completeness other hints and errors on the same lines are also printed.
+            """
+        ),
+    )
 
     parse_parser.set_defaults(func=_parse_command)
 
     parsed = parser.parse_args()
     parsed.func(parsed)
 
 
+ErrorSummary = Dict[str, Dict[str, int]]
+
+
+def _load_json_file(filepath: pathlib.Path) -> Dict[str, Any]:
+    with filepath.open() as json_file:
+        return cast(ErrorSummary, json.load(json_file))
```

Comment:
We shouldn't do the `cast` here anymore.

Reply:
Oh, I thought I'd removed that -- oops. Thanks!